src/client/java/net/neoforged/neoforge/client/event/RegisterDebugEntriesEvent.java

```diff
@@ -0,0 +1,129 @@
+        return entries.containsKey(id);
+    }
+
+    /**
+     * Includes the entry into the given profile.
+     *
+     * @param id            Registration ID for this entry.
+     * @param profile       Debug profile this entry will be included with.
+     * @param profileStatus Status this entry will be set to when the profile is enabled.
+     */
+    public void includeInProfile(ResourceLocation id, DebugScreenProfile profile, DebugScreenEntryStatus profileStatus) {
+        if (getProfileMap(profile).putIfAbsent(id, profileStatus) != null)
+            throw new IllegalStateException("Duplicate DebugScreenEntry " + profile.getSerializedName() + "-profile inclusion: " + id);
+    }
+
+    /**
+     * {@return true if the entry is included into the given profile}
+     */
+    public boolean isIncludedInProfile(ResourceLocation id, DebugScreenProfile profile) {
+        return getProfileMap(profile).containsKey(id);
+    }
+
+    private Map<ResourceLocation, DebugScreenEntryStatus> getProfileMap(DebugScreenProfile profile) {
+        return switch (profile) {
+            case DEFAULT -> defaultProfile;
+            case PERFORMANCE -> performanceProfile;
+        };
+    }
+
+    @ApiStatus.Internal
+    public static void validateProfiles(RegisterDebugEntriesEvent event) {
+        // we delegate validation to its own method to allow people to call 'includeInProfile' before 'register'
+        var defaultError = validateProfile(event, DebugScreenProfile.DEFAULT);
+        var performanceError = validateProfile(event, DebugScreenProfile.PERFORMANCE);
+
+        // we delay throwing the error to allow both profiles to be validated in the same run
+        // we should still validate 'performance' if 'default' is invalid
+        if (defaultError != null && performanceError != null) {
+            var error = new IllegalStateException("Fatal errors occurred while validating DebugScreenProfiles");
+            error.addSuppressed(defaultError);
+            error.addSuppressed(performanceError);
+            throw error;
+        } else if (defaultError != null)
+            throw defaultError;
+        else if (performanceError != null)
+            throw performanceError;
+    }
+
+    @Nullable
+    private static IllegalStateException validateProfile(RegisterDebugEntriesEvent event, DebugScreenProfile profile) {
+        var profileMap = event.getProfileMap(profile);
+        var invalidIds = Sets.difference(profileMap.keySet(), event.entries.keySet());
+
+        if (!invalidIds.isEmpty()) {
+            var logger = LogManager.getLogger();
+            var msg = logger.getMessageFactory().newMessage("Found {} unregistered debug entries in profile: {}", invalidIds.size(), profile.getSerializedName());
+
+            logger.fatal(msg.getFormattedMessage());
+            logger.fatal("Unregistered debug entries: {}", invalidIds.stream().map(ResourceLocation::toString).collect(Collectors.joining(",", "[", "]")));
```

Comment:
Prefer using SLF4J via `LogUtils.getLogger()` and use a `StringBuilder` instead of the logger's message factory